pkg/virt-controller/watch/vm_test.go

```diff
@@ -6461,6 +6451,57 @@ var _ = Describe("VirtualMachine", func() {
 					[]string{virtconfig.VMLiveUpdateFeaturesGate}, &liveUpdate, Not(restartRequiredMatcher(k8sv1.ConditionTrue))),
 			)
 		})
+
+		Context("RunStrategy", func() {
+
+			It("when starting a VM with RerunOnFailure the VM should get started", func() {
+				vm, _ := DefaultVirtualMachine(true)
+				vm.Spec.Running = nil
+				vm.Spec.RunStrategy = kvpointer.P(v1.RunStrategyRerunOnFailure)
+
+				vm, err := virtFakeClient.KubevirtV1().VirtualMachines(vm.Namespace).Create(context.TODO(), vm, metav1.CreateOptions{})
+				Expect(err).To(Succeed())
+				addVirtualMachine(vm)
+
+				sanityExecute(vm)
+
+				vmi, err := virtFakeClient.KubevirtV1().VirtualMachineInstances(vm.Namespace).Get(context.TODO(), vm.Name, metav1.GetOptions{})
+				Expect(err).To(Succeed())
+				Expect(vmi).ToNot(BeNil())
+			})
+
+			DescribeTable("The VM should get started when switching to RerunOnFailure from", func(runStrategy v1.VirtualMachineRunStrategy) {
```

Comment:
Suggestion: RerunOnFailure -> Halted -> RerunOnFailure would be a nice scenario to test.

Reply:
Added.